src/main/kotlin/com/avail/descriptor/representation/AbstractDescriptor.kt

```diff
@@ -0,0 +1,4202 @@
+					{
+						append(", ")
+					}
+					append(f.name)
+					append("=")
+					try
+					{
+						append(f[this])
+					}
+					catch (e: IllegalArgumentException)
+					{
+						append("(inaccessible)")
+					}
+					catch (e: IllegalAccessException)
+					{
+						append("(inaccessible)")
+					}
+				}
+			}
+		}
+		if (fieldCount > 0)
+		{
+			append(")")
+		}
+		return toString()
+	}
+
+	/**
+	 * Throw an
+	 * [unsupported&#32;operation&#32;exception][AvailUnsupportedOperationException]
+	 * suitable to be thrown by the sender.
+	 *
+	 * The exception indicates that the receiver does not meaningfully implement
+	 * the method that immediately invoked this.  This is a strong indication
+	 * that the wrong kind of object is being used somewhere.
+	 *
+	 * @throws AvailUnsupportedOperationException
+	 */
+	fun unsupportedOperation (): Nothing
+	{
+		val callerName =
+			try
+			{
+				throw Exception("just want the caller's frame")
+			}
+			catch (e: Exception)
+			{
+				e.stackTrace[1].methodName
+			}
+		throw AvailUnsupportedOperationException(javaClass, callerName)
+	}
+
+	/**
+	 * Throw an
+	 * [unsupported&#32;operation&#32;exception][AvailUnsupportedOperationException]
+	 * suitable to be thrown by the sender.
+	 *
+	 * The exception indicates that the receiver does not meaningfully implement
+	 * the method that immediately invoked this.  This is a strong indication
+	 * that the wrong kind of object is being used somewhere.
+	 *
+	 * This is a variant on [unsupportedOperation] to support Java
+	 * implementations of [AbstractDescriptor] until porting of the hierarchy is
+	 * finished. It should be thrown by Java callers, which cannot correctly
+	 * reason about flow around non-returning functions.
+	 *
+	 * @return
+	 *   Never returns anything; always throws.
+	 * @throws AvailUnsupportedOperationException
+	 *   Always.
+	 */
+	fun unsupportedOperationException (): AvailUnsupportedOperationException
+	{
+		val callerName =
+			try
+			{
+				throw Exception("just want the caller's frame")
+			}
+			catch (e: Exception)
+			{
+				e.stackTrace[1].methodName
+			}
+		throw AvailUnsupportedOperationException(javaClass, callerName)
+	}
+
+	/**
+	 * Answer whether the [argument types][AvailObject.argsTupleType] supported
+	 * by the specified [function type][FunctionTypeDescriptor] are acceptable
+	 * argument types for invoking a [function][FunctionDescriptor] whose type
+	 * is the self.
+	 *
+	 * @param self
+	 *   A function type.
+	 * @param functionType
+	 *   A function type.
+	 * @return
+	 *   `true` if the arguments of self are, pairwise, more general than those
+	 *   of `functionType`, `false` otherwise.
+	 * @see AvailObject.acceptsArgTypesFromFunctionType
+	 */
+	abstract fun o_AcceptsArgTypesFromFunctionType (
+		self: AvailObject,
+		functionType: A_Type): Boolean
+
+	/**
+	 * Answer whether these are acceptable [argument types][TypeDescriptor] for
+	 * invoking a [function][FunctionDescriptor] whose type is the self.
```

Comment:
-the

Reply:
There are just three bogus ones in `AbstractDescriptor`. All but one other mention of "the self" are about self types, and are appropriate; the last is part of "the self-rescuing princess", which is also appropriate. I fixed the bad ones.